centreon/src/Core/ResourceAccess/Domain/Model/DatasetFilter/ResourceNamesById.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\ResourceAccess\Domain\Model\DatasetFilter;
+
+use Core\Common\Domain\TrimmedString;
+
+class ResourceNamesById
+{
+    /** @var array<int,TrimmedString> */
+    private array $names = [];
+
+    /**
+     * @param int $resourceId
+     * @param TrimmedString $resourceName
+     */
+    public function addName(int $resourceId, TrimmedString $resourceName): void
+    {
+        $this->names[$resourceId] = $resourceName;
+    }
+
+    /**
+     * @param int $resourceId
+     *
+     * @return null|string
+     */
+    public function getName(int $resourceId): ?string
+    {
+        return isset($this->names[$resourceId]) ? $this->names[$resourceId]->value : null;
+    }
+
+    /**
+     * @param array<int,TrimmedString> $names
+     *
+     * @return ResourceNamesById
+     */
+    public function setNames(array $names): self
+    {
+        $this->names = $names;
```

Comment:
As you are not checking the elements of this table, you can do it this way:
```suggestion
        foreach ($names as $id => $name) {
            $this->addName($id, $name);
        }
```